src/model/wallet.js

```diff
@@ -259,37 +439,125 @@ const saveWalletDetails = async (seedPhrase, privateKey, address) => {
   if (canAuthenticate) {
     isSimulator = __DEV__ && (await DeviceInfo.isEmulator());
   }
+  if (canAuthenticate && !isSimulator) {
+    privateAccessControlOptions = {
+      accessControl: ACCESS_CONTROL.USER_PRESENCE,
+      accessible: ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+    };
+  }
 
+  const key = `${address}_${privateKeyKey}`;
+  const val = {
+    address,
+    privateKey,
+    version: privateKeyVersion,
+  };
+
+  await keychain.saveObject(key, val, privateAccessControlOptions);
+};
+
+export const getPrivateKey = async (
+  address,
+  authenticationPrompt = lang.t('wallet.authenticate.please')
+) => {
+  try {
+    const key = `${address}_${privateKeyKey}`;
+    return keychain.loadObject(key, {
+      authenticationPrompt,
+    });
+  } catch (error) {
+    captureException(error);
+    return null;
+  }
+};
+
+export const saveSeedPhrase = async (seedphrase, keychain_id = null) => {
   let privateAccessControlOptions = {};
+  const canAuthenticate = await canImplyAuthentication({
+    authenticationType: AUTHENTICATION_TYPE.DEVICE_PASSCODE_OR_BIOMETRICS,
+  });
+
+  let isSimulator = false;
+
+  if (canAuthenticate) {
+    isSimulator = __DEV__ && (await DeviceInfo.isEmulator());
+  }
   if (canAuthenticate && !isSimulator) {
     privateAccessControlOptions = {
       accessControl: ACCESS_CONTROL.USER_PRESENCE,
       accessible: ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
     };
   }
 
+  const key = `${keychain_id}_${seedPhraseKey}`;
+  const val = {
+    id: keychain_id,
+    seedphrase,
+    version: seedPhraseVersion,
+  };
+
+  await keychain.saveObject(key, val, privateAccessControlOptions);
+};
+
+export const getSeedPhrase = async (
+  id,
+  authenticationPrompt = lang.t('wallet.authenticate.please')
+) => {
+  try {
+    const key = `${id}_${seedPhraseKey}`;
+    return keychain.loadObject(key, {
+      authenticationPrompt,
+    });
+  } catch (error) {
+    captureException(error);
+    return null;
+  }
+};
+
+export const setSelectedWallet = async wallet => {
   const publicAccessControlOptions = {
     accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
   };
 
-  saveSeedPhrase(seedPhrase, privateAccessControlOptions);
-  savePrivateKey(privateKey, privateAccessControlOptions);
-  saveAddress(address, publicAccessControlOptions);
+  const val = {
+    version: selectedWalletVersion,
+    wallet,
+  };
+
+  await keychain.saveObject(selectedWalletKey, val, publicAccessControlOptions);
 };
 
-const saveSeedPhrase = async (seedPhrase, accessControlOptions = {}) => {
-  await keychain.saveString(seedPhraseKey, seedPhrase, accessControlOptions);
+export const getSelectedWallet = async (
+  authenticationPrompt = lang.t('wallet.authenticate.please')
+) => {
+  try {
+    return keychain.loadObject(selectedWalletKey, {
+      authenticationPrompt,
```

Comment:
authentication prompt not needed here

Reply:
We decided to add the param to loadObject